### DISCUSSION TITLE
QMIX Development Timeline Query

### DISCUSSION THREAD

**Discussion by amacrutherford** [instadeepai/Mava#393] [General]
Hi there,

I am looking to use QMIX on a custom PettingZoo environment, I was wondering whether you have a rough eta for its implementation being complete?

Best,
Alex

**Comment by KaleabTessera** [comment]
Hi @am-rutherford :wave: 

So we have a new, improved implementation of qmix (https://github.com/instadeepai/Mava/pull/362), example (https://github.com/instadeepai/Mava/blob/8be1abc34032032154ecb62e26daa492de088ac0/examples/smac/recurrent/centralised/run_qmix.py). 

We are still in the process of fully benchmarking this branch (this might take ~ week). You can work directly of this branch if you would like to use the new implementation before it has been merged in.

**Reply by amacrutherford** [reply]
Hi @KaleabTessera,

Cheers for the reply, I will take a look!
